pkg/main.go

```diff
@@ -1,23 +1,49 @@
 package main
 
 import (
-	"github.com/sarulabs/dingo"
-	"github.com/softonic/homing-pigeon/pkg/generatedServices/dic"
+	"github.com/softonic/homing-pigeon/pkg/ack"
+	"github.com/softonic/homing-pigeon/pkg/messages"
+	"github.com/softonic/homing-pigeon/pkg/middleware"
+	"github.com/softonic/homing-pigeon/pkg/readers"
+	"github.com/softonic/homing-pigeon/pkg/writers"
 	"k8s.io/klog"
+	"os"
+	"strconv"
 )
 
 func main() {
 	klog.InitFlags(nil)
 
-	container, err := dic.NewContainer(dingo.App)
+	bufLen, err := strconv.Atoi(os.Getenv("MESSAGE_BUFFER_LENGTH"))
+	if err != nil {
+		bufLen = 0
+	}
+	inputChannel := make(chan messages.Message, bufLen)
+	outputChannel := make(chan messages.Message, bufLen)
+
+	bufLen, err = strconv.Atoi(os.Getenv("ACK_BUFFER_LENGTH"))
+	if err != nil {
+		bufLen = 0
+	}
+	ackChannel := make(chan messages.Ack, bufLen)
+	brokerAckChannel := make(chan messages.Ack, bufLen)
+	ackManagerChannel := make(chan messages.Ack, bufLen)
+
+	reader, err := readers.NewReader(inputChannel, ackChannel)
 	if err != nil {
 		panic(err)
 	}
-	reader := container.GetReader()
-	middleware := container.GetMiddleware()
-	writer := container.GetWriter()
+
+	writer, err := writers.NewWriter(outputChannel, ackManagerChannel)
+	if err != nil {
+		panic(err)
+	}
+
+	middlewareManager := middleware.NewMiddlewareManager(inputChannel, outputChannel)
+	ackManager := ack.NewAckManager(ackManagerChannel, ackChannel, brokerAckChannel)
```

Comment:
I think we are missing a piece here.
In this case we give `ackManagerChannel` and `ackChannel` (`responseChannel`?) to `ackManager`, and it is its responsability to get the message through.
I think the core program should do this fanout, and the `AckManager` can be opted in if needed. WDYT @paupm ?